weave-js/src/components/Panel2/ChildPanelExportReport/ChildPanelExportReport.tsx

```diff
@@ -36,7 +43,13 @@ export const ChildPanelExportReport = ({
     null
   );
 
+  const selectedReportQuery = useQuery(GET_REPORT, {
```

Comment:
I know you are using this right now for example purposes and the main thing is to get mutations working. We should probably avoid GQL queries in weave components as much as possible. Once you are done with the mutation side, it would be nice to see if this can be articulated as a weave query.

Reply:
Ah, I did work on a version of this request using weave queries at one point. I can refactor if you feel strongly about it, but truthfully I just find the direct GQL queries so much more intuitive. I know you said there are perf benefits to weave queries, but it's really hard to resist the GQL option when you consider dev experience/velocity.

I was under the impression that weave queries are only required in the context of panels (to execute input nodes), and that it would be the dev's discretion whether to use gql queries or weave queries outside of that -- as long as the two approaches are not mixed?

To be honest, I think most folks coming over from core will opt for the gql approach if it's available. The barrier is just so much lower than learning how to construct weave nodes. It might be difficult to enforce the preference for weave queries once we merge this PR.

Reply:
totally fair perspective. I believe that developers should have the power to choose what is best/easiest. My only comment on why you might NOT use the GQL approach for certain things is that if you have some thing like this:

```
Component A (assume calls useNodeValue to get some id to pass to B)
Component B (use ID in GQL query to fetch some other ID)
Component C (use ID to construct weave graph, then call useNodeValue)
```

In this case, let's say we have a `Loader` for each component... we will get a waterfall effect - where B does not load till A is done, then C doesn't load till B is done, then C needs to load the data - this is 3 serialized requests (aka waterfall problem).

Alternatively, if this was refactored such that:

```
Component A (passes weave node to B)
Component B (adds ops to incoming node, passing result to C)
Component C (adds ops to incoming node, then calls useNodeValue)
```

In this case, there is no loading state or network calls until C loads.

---

In conclusion here, i think it is good to add this client, and agree that it is easier in some cases. However, I think it is important to call out that this does allow users to have a "performance foot gun" if they design parts of their app like the above.